pkg/cloud/services/ec2/instances.go

```diff
@@ -81,10 +81,6 @@ func (s *Service) InstanceIfExists(id *string) (*v1alpha2.Instance, error) {
 
 	input := &ec2.DescribeInstancesInput{
 		InstanceIds: []*string{id},
-		Filters: []*ec2.Filter{
-			filter.EC2.VPC(s.scope.VPC().ID),
-			filter.EC2.InstanceStates(ec2.InstanceStateNamePending, ec2.InstanceStateNameRunning),
-		},
```

Comment:
Removing this filter is likely to result in issues with trying to recreate a Machine with the same name after deletion if the backing instance is still in a Terminated state.

Reply:
The logic of checking the state of a Machine moved up now, where we set the error state if the Machine is not running or pending.

Reply:
Right, so if a user deletes a Machine and the backing instance is terminated and then the user tries to create a new Machine with the same name, this will now return the terminated instance and result in the new Machine returning an error rather than creating a new instance.

Reply:
In case a Machine is in a terminated state, after the we check the ProviderID, if we don't have one (which means we can't query by ID), we query by tags using `InstanceByTags`, which still has the filtering behavior you'd expect. We should probably rename the method to something like `GetRunningInstanceByTags`